app/server/lib/DocManager.ts

```diff
@@ -454,7 +464,31 @@ export class DocManager extends EventEmitter {
     return this._activeDocs.get(docName);
   }
 
+  /**
+   * ActiveDoc uses this to register the SQLiteDB associated with it,
+   * when there is one. It might seem easier just to get it from
+   * activeDoc.docStorage when you need it, but you can end
+   * up in a loop or hung if you are checking during document
+   * initialization.
+   */
+  public registerSQLiteDB(docName: string, db?: SQLiteDB) {
```

Comment:
I think splitting this into `registerSQLiteDB` and `unregisterSQLiteDB` might be better. The former reads a little funny in cleanup contexts like `removeActiveDoc`.